dask/dataframe/tests/test_multi.py

```diff
@@ -1072,6 +1072,29 @@ def check_and_return(ddfs, dfs, join):
         assert has_known_categories(res.y) == known
 
 
+def test_concat_datetimeindex():
+    # https://github.com/dask/dask/issues/2932
+    b2 = pd.DataFrame({'x': ['a']},
+                      index=pd.DatetimeIndex(['2015-03-24 00:00:16'],
+                                             dtype='datetime64[ns]'))
+    b3 = pd.DataFrame({'x': ['c']},
+                      index=pd.DatetimeIndex(['2015-03-29 00:00:44'],
+                                             dtype='datetime64[ns]'))
+
+    b2['x'] = b2.x.astype('category').cat.set_categories(['a', 'c'])
+    b3['x'] = b3.x.astype('category').cat.set_categories(['a', 'c'])
+
+    db2 = dd.from_pandas(b2, 1)
+    db3 = dd.from_pandas(b3, 1)
+
+    result = concat([db2._meta, db3._meta])
```

Comment:
Small nitpick, this is testing that concatenating empty frames doesn't change the index type, but relies on `_meta` being an empty frame (which is an implementation detail). Slight preference to change to:

```python
result = concat([b2.iloc[:0], b3.iloc[:0]])
```